components/calendar/src/chinese.rs

```diff
@@ -0,0 +1,1044 @@
+        while next_month <= date && month < max_months {
+            month += 1;
+            cur_month = next_month;
+            next_month = Self::chinese_new_moon_on_or_after((cur_month + 1).as_moment());
+        }
+        debug_assert!(month < max_months, "Unexpectedly large number of months");
+        let day = (date - cur_month + 1) as u8;
+        Date::try_new_chinese_date(year, month, day).unwrap()
+    }
+
+    /// Get a RataDie from a ChineseDateInner
+    pub(crate) fn fixed_from_chinese_date_inner(date: ChineseDateInner) -> RataDie {
+        let year = date.0.year;
+        let month = date.0.month as i64;
+        let day = date.0.day as i64;
+        let mid_year = Self::fixed_mid_year_from_year(year);
+        let new_year = Self::chinese_new_year_on_or_before_fixed_date(mid_year);
+        let month_approx = new_year + (month - 1) * 29;
+        let prior_new_moon = Self::chinese_new_moon_on_or_after(month_approx.as_moment());
+        prior_new_moon + day - 1
+    }
+
+    /// Get the mid-year RataDie of a given year
+    fn fixed_mid_year_from_year(elapsed_years: i32) -> RataDie {
+        let cycle = quotient(elapsed_years - 1, 60) + 1;
+        let year = adjusted_rem_euclid(elapsed_years, 60);
+        CHINESE_EPOCH + ((((cycle - 1) * 60 + year - 1) as f64 + 0.5) * MEAN_TROPICAL_YEAR) as i64
+    }
+
+    /// Returns true if the fixed date given is in a leap year, false otherwise
+    fn fixed_date_is_in_leap_year(date: RataDie) -> bool {
+        let prev_new_year = Self::chinese_new_year_on_or_before_fixed_date(date);
+        let next_new_year = Self::chinese_new_year_on_or_before_fixed_date(prev_new_year + 400);
+        let difference = next_new_year - prev_new_year;
+        difference > 365
+    }
+
+    /// Given that a date is in a leap year, find which month in the year is a leap month.
+    /// Since the first month in which there are no major solar terms is a leap month,
+    /// this function cycles through months until it finds the leap month, then returns
+    /// the number of that month. This function assumes the date passed in is in a leap year
+    /// and tests to ensure this is the case by asserting that no more than twelve months are
+    /// analyzed.
+    fn get_leap_month_in_year(date: RataDie) -> u8 {
+        let mut cur = Chinese::chinese_new_year_on_or_before_fixed_date(date);
+        let mut result = 1;
+        let max_iters = 13;
+        while result < max_iters && !Self::chinese_no_major_solar_term(cur) {
+            cur = Chinese::chinese_new_moon_on_or_after((cur + 1).as_moment());
+            result += 1;
+        }
+        debug_assert!(result < max_iters, "The given year was not a leap year and an unexpected number of iterations occurred searching for a leap month.");
+        result
+    }
+
+    /// Get a FormattableYear from an integer Chinese year
+    fn format_chinese_year(year: i32) -> FormattableYear {
+        let era = Era(tinystr!(16, "era"));
```

Comment:
issue: era name for this calendar should just be `"chinese"`

Reply:
Done